forwarder/internal/logs/parse.go

```diff
@@ -35,55 +35,83 @@ type Parser interface {
+	err := json.Unmarshal(bytes, &flowEventRecords)
+	return &flowEventRecords, err
+}
+
+func processFlowEventRecords[T flowEventRecord](flowEventRecords *flowEventRecords[T], blob storage.Blob, originalSize int, scrubbedSize int, piiScrubber Scrubber, yield func(ParsedLogResponse) bool) bool {
+	response := ParsedLogResponse{}
+	for idx, flowEventLog := range flowEventRecords.Records {
+		currLog, err := flowEventLog.ToLog(blob)
+		response.ParsedLog = currLog
+		if err != nil {
+			response.Err = err
+			yield(response)
+			return false
+		}
+		if idx == len(flowEventRecords.Records)-1 {
+			response.ParsedLog.RawByteSize = int64(originalSize)
+			response.ParsedLog.ScrubbedByteSize = int64(scrubbedSize)
+		}
+
+		if !yield(response) {
+			return false
+		}
+	}
+	return true
+}
+
 // Parse reads logs from a reader and parses them into Log objects.
 func (f FlowEventParser) Parse(scanner *bufio.Scanner, blob storage.Blob, piiScrubber Scrubber) iter.Seq[ParsedLogResponse] {
 	return func(yield func(ParsedLogResponse) bool) {
 		for scanner.Scan() {
 			currBytes := scanner.Bytes()
-			var flowLogs vnetFlowLogs
 			originalSize := len(currBytes)
 			scrubbedBytes := piiScrubber.Scrub(currBytes)
-
+			scrubbedSize := len(scrubbedBytes)
 			response := ParsedLogResponse{}
 
-			err := json.Unmarshal(scrubbedBytes, &flowLogs)
-			if err != nil {
-				response.Err = err
-				yield(response)
-				return
-			}
-			for idx, flowLog := range flowLogs.Records {
-				currLog, err := flowLog.ToLog(blob)
+			switch blob.Container.Name {
+			case NetworkSecurityGroupFlowEventContainer:
+				networkSecGroupRecords, err := unmarshalFlowEventRecords[*networkSecurityGroupFlowLog](scrubbedBytes)
 				if err != nil {
 					response.Err = err
 					yield(response)
 					return
 				}
-				if idx == len(flowLogs.Records)-1 {
-					currLog.RawByteSize = int64(originalSize)
-				}
-				response.ParsedLog = currLog
-				if !yield(response) {
+				processFlowEventRecords[*networkSecurityGroupFlowLog](networkSecGroupRecords, blob, originalSize, scrubbedSize, piiScrubber, yield)
+			case FlowEventContainer:
+				vnetFlowRecords, err := unmarshalFlowEventRecords[*vnetFlowEventLog](scrubbedBytes)
+				if err != nil {
+					response.Err = err
+					yield(response)
 					return
 				}
+				processFlowEventRecords[*vnetFlowEventLog](vnetFlowRecords, blob, originalSize, scrubbedSize, piiScrubber, yield)
+			default:
+				response.Err = errors.New("no parser found for log type" + blob.Container.Name)
+				yield(response)
+				return
 			}
-			continue
-
 		}
 	}
 }
 
-var flowEventContainers = []string{
-	"insights-logs-flowlogflowevent",
-	"insights-logs-networksecuritygroupflowevent",
-}
+const (
+	NetworkSecurityGroupFlowEventContainer = "insights-logs-networksecuritygroupflowevent"
+	FlowEventContainer                     = "insights-logs-flowlogflowevent"
+)
+
+var vnetFlowEventContainers = []string{FlowEventContainer, NetworkSecurityGroupFlowEventContainer}
```

Comment:
this still has the naming issue where the joint thing is described as the discrete thing and the discrete thing is described as a joint thing. 
this should be
```
const (
	NetworkSecurityGroupFlowEventContainer = "insights-logs-networksecuritygroupflowevent"
	VnetFlowEventContainer                     = "insights-logs-flowlogflowevent"
)

var flowEventContainers = []string{VnetFlowEventContainer, NetworkSecurityGroupFlowEventContainer}
```